kolibri/plugins/coach/assets/src/routes/index.js

```diff
@@ -58,10 +70,11 @@ export default [
     name: PageNames.HOME_PAGE,
     path: '/:classId?/home',
     component: HomePage,
-    handler: (toRoute, fromRoute, next) => {
+    handler: async (toRoute, fromRoute, next) => {
```

Comment:
I think that here `async` could be removed because in this function we're not `await`ing, or was there some trouble without it? Same on line 87.

Reply:
Yes I tried this approach but it was not able to load the promises properly. So, I added async here.

Reply:
I am not sure if I can understand what exactly you mean "it was not able to load the promises properly".

If I remove `async` from here, it doesn't throw any errors and works like before in the current code, because we're not calling `await showHomePage(toRoute)`. However, because of that missing `await`, I don't think that promises get resolved before `'notLoading'`.

Overall looking at the way you implemented it, I believe the intention actually was to wait for these two promises to be finished before calling `'notLoading'`, which we indeed need here. To achieve that, we could:

```
function showHomePage(toRoute) {
  const initClassInfoPromise = store.dispatch('initClassInfo', toRoute.params.classId);
  const getFacilitiesPromise =
    store.getters.isSuperuser && store.state.core.facilities.length === 0
      ? store.dispatch('getFacilities').catch(() => {})
      : Promise.resolve();

  return Promise.all([initClassInfoPromise, getFacilitiesPromise]);
}
```

and then

```
{
    name: PageNames.HOME_PAGE,
    path: '/:classId?/home',
    component: HomePage,
    handler: async (toRoute, fromRoute, next) => {
      if (classIdParamRequiredGuard(toRoute, HomePage.name, next)) {
        return;
      }
      await showHomePage(toRoute);
      store.dispatch('notLoading');
    },
```

Is this what you've meant to achieve?

Reply:
I think this is one last thing to cleanup - seems to be caused by playing around with two possible placements of `store.dispatch('notLoading');` that got mixed up. Since now it's removed from the helper `showHomePage` function, we need to update the structure a bit.